### PR TITLE
fix(sidebar): make chat purpose text selectable and fix ability to break layout when editing purpose

### PR DESCRIPTION
**Pull request by lostfictions**
Currently the sidebar purpose editor can be resized in a way that breaks layout:

screenshot from 2017-11-08 18-00-45

This PR fixes that, and also makes chat purpose text selectable (so stuff like URLs in the purpose can at least be copy-pasted even if they're not clickable yet)

### CODE REVIEW

File: src/ui/chat/components/sidebar/ChannelSideBar.jsx [hunk 1]
```diff
@@ -81,8 +81,14 @@ class ChannelSideBar extends React.Component {
                         <div onClick={this.showChatPurposeEditor}>
                             {
                                 this.chatPurposeEditorVisible
-                                    ? <ChatNameEditor showLabel tabIndex="-1" purpose readOnly={!canIAdmin}
-                                        onBlur={this.hideChatPurposeEditor} multiline
+                                    ? <ChatNameEditor
```
Comment (anri-asaturov) [inline src/ui/chat/components/sidebar/ChannelSideBar.jsx]: I see you like long code files :-D

**Comment by anri-asaturov** [comment]
Did you test this enough or shall i summon Eren?

File: src/ui/chat/components/sidebar/ChannelSideBar.jsx (hunk 1 above)
Comment (lostfictions) [reply src/ui/chat/components/sidebar/ChannelSideBar.jsx]: you're right! in fact, i think we should also adopt object syntax like this
```
{ foo, bar,
  baz: 'value', barfly: true,
  gorf: this.bork, garple,
  dorg
}
```
:rofl:

**Comment by lostfictions** [comment]
Maybe if he were up for taking a quick look that'd be a good idea? It shouldn't break anything but I didn't test super thoroughly

File: src/ui/chat/components/sidebar/ChannelSideBar.jsx (hunk 1 above)
Comment (anri-asaturov) [reply src/ui/chat/components/sidebar/ChannelSideBar.jsx]: Absolutely hate it )

**Comment by anri-asaturov** [comment]
@ebilg the scene is yours

**Comment by lostfictions** [comment]
Actually, I just took another quick look at this and tried to break it -- I don't think Eren necessarily needs to look at it since there's no real behaviour changes other than the resize grippy being gone. Should be ok to merge.

**Pull request merged**